batchflow/models/torch/base.py

```diff
@@ -1847,3 +1853,87 @@ def _parse_profilers(self):
                                          columns=['ncalls', 'CPU_tottime', 'CPU_cumtime', 'CUDA_cumtime'])
         self.profile_info['CPU_tottime_avg'] = self.profile_info['CPU_tottime'] / self.profile_info['ncalls']
         self.profile_info['CUDA_cumtime_avg'] = self.profile_info['CUDA_cumtime'] / self.profile_info['ncalls']
+
+
+    # Utilities for activations
+    @staticmethod
+    def get_blocks_and_activations(model, modules=None):
+        """ Retrieve intermediate blocks of the neural network model
+        and corresponding activation names.
+
+        Parameters
+        ----------
+        model : Network
+
+        modules : str or list of str, default None
+            The modules of the model for retrieving activations from.
+            If None, all of the parts from the `model.config['order']` will be used.
+
+        Returns
+        -------
+        activation_blocks : list of str
+            Intermediate activation blocks.
+        activation_names : list of str
+            Names of activations corresponding to activation blocks.
+
+        Examples
+        --------
+        >>> get_blocks_and_activations(model, modules=['encoder', 'embedding', 'decoder'])
+        >>> print(activation_blocks)
+        ['model.encoder["block-0"]', 'model.encoder["block-1"]', 'model.embedding', 'model.decoder["block-0"]']
+        >>> print(activation_names)
+        ['encoder_0', 'encoder_1', 'embedding_0', 'decoder_0']
+        """
+        modules = [modules] if isinstance(modules, str) else modules or model.config['order']
+        activation_names = []
+        activation_blocks = []
+
+        for module in modules:
+            number = -1
+            extracted_module = getattr(model, module)
```

Comment:
Try to not mix naming schemes. Here, `module` is a `str`, but `extracted_module` is a `nn.Module`. Use `module_name` instead.